### PR TITLE
saving vehicle to the database

### PR DESCRIPTION
**Pull request by NietThijmen**
## Description

Adds the function of saving a vehicle to the database, as that is not in the current version of this script.

## Checklist

- [ ] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

File: server/server.lua [hunk 1]
```diff
@@ -6,5 +6,21 @@ end)
...
-end)
\ No newline at end of file
+    if IsVehicleOwned(plate) then 
+        exports.oxmysql:execute('UPDATE player_vehicles SET mods = ? WHERE plate = ?', {json.encode(mods), plate})
+    end
+end)
+
+
+
+function IsVehicleOwned(plate)
```
Comment (BerkieBb) [inline server/server.lua]: Local function please

File: server/server.lua [hunk 2]
```diff
@@ -6,5 +6,21 @@ end)
 
 RegisterNetEvent("qbx-customs:server:buyCart", function(cart)
     local src = source
+    
+    local plate = cart.plate
+    local mods = cart.mods
 
-end)
\ No newline at end of file
+    if IsVehicleOwned(plate) then
```
Comment (BerkieBb) [inline server/server.lua]: Invert if statement

File: client/client.lua [hunk 3]
```diff
@@ -45,8 +45,17 @@ RegisterNUICallback("UpdateCart", function(data, cb)
...
+    local veh = GetVehiclePedIsIn(cache.ped, false)
+    local mods = QBCore.Functions.GetVehicleProperties(veh)
+    local plate = GetVehicleNumberPlateText(veh)
+
+    local sendData = {
+        plate = plate,
+        mods = mods
+    }
+    
+    TriggerServerEvent('xv-customs:server:buyCart', sendData)
```
Comment (BerkieBb) [inline client/client.lua]: Event name doesn't align

File: client/client.lua [hunk 4]
```diff
@@ -45,8 +45,17 @@ RegisterNUICallback("UpdateCart", function(data, cb)
 end)
 
 RegisterNUICallback("PurchaseCart", function(data, cb)
-    -- PurchaseCart(data)
-    TriggerServerEvent('xv-customs:server:buyCart', data)
+    local veh = GetVehiclePedIsIn(cache.ped, false)
+    local mods = QBCore.Functions.GetVehicleProperties(veh)
+    local plate = GetVehicleNumberPlateText(veh)
```
Comment (BerkieBb) [inline client/client.lua]: Use QBCore.Functions.GetPlate instead

File: client/client.lua [hunk 5]
```diff
@@ -45,8 +45,17 @@ RegisterNUICallback("UpdateCart", function(data, cb)
 end)
 
 RegisterNUICallback("PurchaseCart", function(data, cb)
-    -- PurchaseCart(data)
-    TriggerServerEvent('xv-customs:server:buyCart', data)
+    local veh = GetVehiclePedIsIn(cache.ped, false)
```
Comment (BerkieBb) [inline client/client.lua]: Use cache.vehicle instead of the native

**Review by BerkieBb — CHANGES_REQUESTED** [review]

**Comment by NietThijmen** [comment]
Thanks Berkie, didn't notice a few of those things

File: server/server.lua [hunk 6]
```diff
@@ -6,5 +6,21 @@ end)
...
+
+
+
+local function IsVehicleOwned(plate)
+    local result = exports.oxmysql:scalarSync('SELECT plate FROM player_vehicles WHERE plate = ?', {plate})
+    if result then
+        return true
+    end
+    return false
+end
```
Comment (Manason) [inline server/server.lua]: Would prefer these database queries to be added to qb-core/server/storage.lua and then exported to be called here, but this is fine for now.

**Comment by BerkieBb** [comment]
A local function must be defined above it's usage, otherwise you'll get an error that it tried to call a nil value

File: server/server.lua [hunk 7]
```diff
@@ -6,5 +6,21 @@ end)
...
 RegisterNetEvent("qbx-customs:server:buyCart", function(cart)
     local src = source
+    
+    local plate = cart.plate
+    local mods = cart.mods
 
-end)
\ No newline at end of file
+    if not IsVehicleOwned(plate) then return end
+    exports.oxmysql:execute('UPDATE player_vehicles SET mods = ? WHERE plate = ?', {json.encode(mods), plate})
```
Comment (Manason) [inline server/server.lua]:
```suggestion
    MySQL.update('UPDATE player_vehicles SET mods = ? WHERE plate = ?', {json.encode(mods), plate})
```

File: server/server.lua [hunk 8]
```diff
@@ -6,5 +6,21 @@ end)
...
+    
+end)
+
+
+
+local function IsVehicleOwned(plate)
+    local result = exports.oxmysql:scalarSync('SELECT plate FROM player_vehicles WHERE plate = ?', {plate})
```
Comment (Manason) [inline server/server.lua]:
```suggestion
    local result = MySQL.scalar.await('SELECT plate FROM player_vehicles WHERE plate = ?', {plate})
```

**Review by Manason — APPROVED** [review]

File: server/server.lua [hunk 9]
```diff
@@ -6,5 +6,21 @@ end)
...
-end)
\ No newline at end of file
+    if not IsVehicleOwned(plate) then return end
+    exports.oxmysql:execute('UPDATE player_vehicles SET mods = ? WHERE plate = ?', {json.encode(mods), plate})
+    
+end)
+
+
+
+local function IsVehicleOwned(plate)
```
Comment (Manason) [inline server/server.lua]:
```suggestion
local function isVehicleOwned(plate)
```
camelCase local functions

**Review by Manason — CHANGES_REQUESTED** [review]

File: fxmanifest.lua [hunk 10]
```diff
@@ -23,9 +23,11 @@ client_script {
   'client/functions.lua',
   'client/client.lua',
 }
+
 server_script {
+  '@mysql-async/lib/MySQL.lua',
```
Comment (BerkieBb) [inline fxmanifest.lua]: oxmysql

File: server/server.lua [hunk 11]
```diff
@@ -1,26 +1,26 @@
...
+    end
+    return false
+end
 
 RegisterServerEvent('qbx-customs:server:buyRepair', function(data)
     local src = source
     TriggerClientEvent("qbx-customs:client:repairVehicle", src)
 end)
 
+
```
Comment (BerkieBb) [inline server/server.lua]: Random open space

**Pull request merged**